addons/mail_plugin/controllers/mail_plugin.py

```diff
@@ -0,0 +1,290 @@
+        # clicking on a contact.
+        filter_domain = ['|', '|', ('display_name', 'ilike', search_term), ('ref', '=', search_term),
+                         ('email', 'ilike', search_term)]
+        # Search for the partner based on the email.
+        # If multiple are found, take the first one.
+        partners = request.env['res.partner'].search(filter_domain, limit=limit)
+
+        partners = [
+            self._prepare_partner_values(partner)
+            for partner in partners
+        ]
+        return {"partners": partners}
+
+    @http.route(['/mail_client_extension/partner/create', '/mail_plugin/partner/create'],
+                type="json", auth="outlook", cors="*")
+    def res_partner_create(self, email, name, company):
+        """
+        params email: email of the new partner
+        params name: name of the new partner
+        params company: parent company id of the new partner
+        """
+        # old route name "/mail_client_extension/partner/create is deprecated as of saas-14.3,it is not needed for newer
+        # versions of the mail plugin but necessary for supporting older versions
+        # TODO search the company again instead of relying on the one provided here?
+        # Create the partner if needed.
+        partner_info = {
+            'name': name,
+            'email': email,
+        }
+
+        #see if the partner has a parent company
+        if company and company > -1:
+            partner_info['parent_id'] = company
+        partner = request.env['res.partner'].create(partner_info)
+
+        response = {'id': partner.id}
+        return response
+
+    @http.route('/mail_plugin/log_mail_content', type="json", auth="outlook", cors="*")
+    def log_mail_content(self, model, res_id, message):
+        if model not in self._mail_content_logging_models_whitelist():
+            raise Forbidden()
+        request.env[model].browse(res_id).message_post(body=message)
+
+    def _iap_enrich(self, domain):
+        enriched_data = {}
+        try:
+            response = request.env['iap.enrich.api']._request_enrich({domain: domain})  # The key doesn't matter
+        except iap_tools.InsufficientCreditError:
+            enriched_data['enrichment_info'] = {'type': 'insufficient_credit',
+                                                'info': request.env['iap.account'].get_credits_url('reveal')}
+        except Exception:
+            enriched_data["enrichment_info"] = {'type': 'other', 'info': 'Unknown reason'}
+        else:
+            enriched_data = response.get(domain)
+            if not enriched_data:
+                enriched_data = {'enrichment_info': {'type': 'no_data',
+                                                     'info': 'The enrichment API found no data for the email provided.'}}
+        return enriched_data
+
+    def _find_existing_company(self, normalized_email):
+        sender_domain = normalized_email.split('@')[1]
```

Comment:
You may use tools.email_domain_extract

By the way, why not always using this tool in your code ?

Reply:
Done.